pkg/tfbridge/x/tokens.go

```diff
@@ -484,13 +484,30 @@ func AutoAliasing(providerInfo *b.ProviderInfo, artifact b.ProviderMetadata) err
 		return err
 	}
 
-	currentVersion := -1
+	var currentVersion int
+	// If version is missing, we assume the most recent major version in history
 	if providerInfo.Version != "" {
 		v, err := semver.NewVersion(providerInfo.Version)
 		if err != nil {
 			return err
 		}
 		currentVersion = int(v.Major())
+	} else {
+		for _, r := range hist.Resources {
+			for _, p := range r.Past {
+				if p.MajorVersion > currentVersion {
+					currentVersion = p.MajorVersion
+				}
+			}
+		}
+		for _, d := range hist.DataSources {
+			for _, p := range d.Past {
+				if p.MajorVersion > currentVersion {
+					currentVersion = p.MajorVersion
+				}
+			}
+		}
+		providerInfo.Version = fmt.Sprintf("%d.0.0", currentVersion)
```

Comment:
Not sure if we want to default to this, or leave empty.

Reply:
I would say leave it empty. We can't guarantee that the version we have computed is correct. Potentially false information is much more confusing then no information here.